oligo_designer_toolsuite/oligo_specificity_filter/_filter_exact_matches.py

```diff
@@ -45,35 +50,45 @@ def apply(self, sequence_type: _TYPES_SEQ, oligo_database: OligoDatabase, n_jobs
         ), f"Sequence type not supported! '{sequence_type}' is not in {options}."
 
         # extract all the sequences
-        search_results = self._get_duplicated_sequences(sequence_type, oligo_database)
+        sequences = oligo_database.get_sequence_list(sequence_type=sequence_type)
+        search_results = self._get_duplicated_sequences(sequences)
 
         # get mapping from sequence to oligo_id
+        # use same sequence as sequence_type
         sequence_oligoids_mapping = oligo_database.get_sequence_oligoid_mapping(
             sequence_type=sequence_type, sequence_to_upper=True
         )
 
         region_ids = list(oligo_database.database.keys())
-        results = Parallel(n_jobs=n_jobs)(
+        table_hits = Parallel(n_jobs=n_jobs)(
             delayed(self._run_filter)(
                 sequence_type=sequence_type,
                 region_id=region_id,
                 oligo_database=oligo_database,
                 search_results=search_results,
                 sequence_oligoids_mapping=sequence_oligoids_mapping,
+                consider_hits_from_input_region=False,
             )
             for region_id in region_ids
         )
 
-        # Process results
-        table_hits, oligos_with_hits = zip(*results)
+        if self.policy:
```

Comment:
I think we should have a dedicated policy that simply deletes both oligos, which would be the default.